app/src/main/java/org/jellyfin/mobile/fragment/WebViewFragment.kt

```diff
@@ -157,19 +158,14 @@ class WebViewFragment : Fragment(), NativePlayerHost {
                 val url = request.url
                 val path = url.path?.lowercase(Locale.ROOT) ?: return null
                 return when {
-                    path.endsWith(Constants.WEB_CONFIG_PATH) -> {
-                        runOnUiThread {
-                            webView.evaluateJavascript(JS_INJECTION_CODE) {
-                                onConnectedToWebapp()
-                            }
-                        }
-                        null // continue loading normally
-                    }
-                    path.contains("native") -> webView.context.loadAsset("native/${url.lastPathSegment}")
-                    path.endsWith(Constants.CAST_SDK_PATH) -> {
-                        // Load the chrome.cast.js library instead
-                        webView.context.loadAsset("native/chrome.cast.js")
+                    path.matches(Constants.MAIN_BUNDLE_PATH_REGEX) && url.query.isNullOrEmpty() -> {
+                        onConnectedToWebapp()
+                        assetsPathHandler.handle("native/injectionScript.js")
                     }
```

Comment:
I'm not a fan of the regex, but this is the most robust implementation I could come up with. Feel free to suggest different methods.